StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -229,52 +229,141 @@ extension PaymentSheet {
             return paymentTypes
         }
 
+        /// Returns whether or not PaymentSheet, with the given `PaymentMethodRequirementProvider`s, should make the given `paymentMethod` available to add.
+        /// Note: This doesn't affect the availability of saved PMs.
+        /// - Parameters:
+        ///   - paymentMethod: the `STPPaymentMethodType` in question
+        ///   - requirementProviders: a list of [PaymentMethodRequirementProvider] who satisfy payment requirements
+        ///   - intent: a intent object
+        ///   - supportedPaymentMethods: the payment methods that PaymentSheet can display UI for
+        /// - Returns: true if `paymentMethod` should be available in the PaymentSheet, false otherwise
         static func supportsAdding(
             paymentMethod: PaymentMethodType,
             configuration: PaymentSheet.Configuration,
             intent: Intent,
             supportedPaymentMethods: [STPPaymentMethodType] = PaymentSheet.supportedPaymentMethods
         ) -> Bool {
-            if let stpPaymentMethodType = paymentMethod.stpPaymentMethodType {
-                return PaymentSheet.supportsAdding(
-                    paymentMethod: stpPaymentMethodType,
-                    configuration: configuration,
-                    intent: intent,
-                    supportedPaymentMethods: supportedPaymentMethods
-                )
-            } else if case .dynamic = paymentMethod {
-                return supports(
-                    requirements: paymentMethod.supportsAddingRequirements(),
+            guard let stpPaymentMethodType = paymentMethod.stpPaymentMethodType else {
+                if case .dynamic = paymentMethod {
+                    return supports(
+                        requirements: paymentMethod.supportsAddingRequirements(),
+                        configuration: configuration,
+                        intent: intent
```

Comment:
@wooj-stripe / @porter-stripe - Can you explain this code and add it as a comment?

The more I look at it, the more confused I get.  I think a dynamic PM means that the type, its form spec, etc. is not defined on the client, and instead comes from the server.  I'd expect the client therefore *doesn't filter* dynamic PMs, since the client doesn't have knowledge of them or its requirements.

But then I see we actually have defined requirements for some dynamic PMs client-side(line 18 in this file), and we filter out any unknown dynamic PMs!  This seems like it'd defeat the whole purpose of dynamic PMs - they still require the client to hardcode support it...

So - what are dynamic PMs and why are we filtering them on the client?

Reply:
Similarly I don't quite understand it, my goal here was to preserve the existing functionality without breaking anything. But I don't grasp it well enough to add comments.

Reply:
> I think a dynamic PM means that the type, its form spec, etc. is not defined on the client, and instead comes from the server.
- Correct

> I'd expect the client therefore doesn't filter dynamic PMs, since the client doesn't have knowledge of them or its requirements.
- Unfortunately, this is not the case, and clients must have the knowledge of a payment method and their requirements  because LUXE never finished having server side support for knowing what version of the client supports which payment method.  Therefore, what you're looking at is a half-baked solution which allows for payment methods that don't have bindings support, to be shown in payment sheet :(.

Reply:
Can we add a long comment with all this info @porter-stripe?

@wooj-stripe Any recommendation on what we do with this "half-baked" solution going forward? Do we plan to finish this LUXE work?

Reply:
Given that LUXE is winding down, and the feature of "dynamically delivering new LPMS to versions of clients that don't know about it" not being a priority (and therefore won't be staffed), I don't for see this being completed in the short nor long term.

I think the best we can do at this point is refactor the code on the client such that it supports the different use cases we support:
* LPM Bindings support + LPM payment sheet support
* LPM payment sheet support only
* LPM Binding support only